packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -767,46 +767,44 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
   // Handles shortcut functionality including cut, copy, paste and select all
   // using control/command + (X, C, V, A).
   Future<void> _handleShortcuts(LogicalKeyboardKey key) async {
+    final TextSelection selection = textSelectionDelegate.textEditingValue.selection;
+    final String text = textSelectionDelegate.textEditingValue.text;
     assert(selection != null);
     assert(_shortcutKeys.contains(key), 'shortcut key $key not recognized.');
     if (key == LogicalKeyboardKey.keyC) {
-      if (!selection!.isCollapsed) {
+      if (!selection.isCollapsed) {
         Clipboard.setData(
-            ClipboardData(text: selection!.textInside(_plainText)));
```

Comment:
Would it be useful to add a comment to the _plainText getter to warn against using it like this again in the future?

Reply:
Definitely recommended!